miio/walkingpad.py

```diff
@@ -0,0 +1,275 @@
+        """True if the device is turned on."""
+        return self.power == "on"
+
+    @property
+    def walking_time(self) -> timedelta:
+        """Current walking duration in seconds."""
+        return int(self.data["time"])
+
+    @property
+    def speed(self) -> float:
+        """Current speed."""
+        return float(self.data["sp"])
+
+    @property
+    def start_speed(self) -> float:
+        """Current start speed."""
+        return self.data["start_speed"]
+
+    @property
+    def mode(self) -> OperationMode:
+        """Current mode."""
+        return OperationMode(self.data["mode"])
+
+    @property
+    def sensitivity(self) -> OperationSensitivity:
+        """Current sensitivity."""
+        return OperationSensitivity(self.data["sensitivity"])
+
+    @property
+    def step_count(self) -> int:
+        """Current steps."""
+        return int(self.data["step"])
+
+    @property
+    def distance(self) -> int:
+        """Current distance in meters."""
+        return int(self.data["dist"])
+
+    @property
+    def calories(self) -> int:
+        """Current calories burnt."""
+        return int(self.data["cal"])
+
+
+class Walkingpad(Device):
+    """Main class representing Xiaomi Walkingpad."""
+
+    @command(
+        default_output=format_output(
+            "",
+            "Power: {result.power}\n"
+            "Mode: {result.mode.name}\n"
+            "Time: {result.walking_time}\n"
+            "Steps: {result.step_count}\n"
+            "Speed: {result.speed}\n"
+            "Start Speed: {result.start_speed}\n"
+            "Sensitivity: {result.sensitivity.name}\n"
+            "Distance: {result.distance}\n"
+            "Calories: {result.calories}",
+        )
+    )
+    def status(self) -> WalkingpadStatus:
+        """Retrieve properties."""
+
+        data = self._get_quick_status()
+
+        # The quick status only retrieves a subset of the properties. The rest of them are retrieved here.
+        properties_additional = ["power", "mode", "start_speed", "sensitivity"]
+        values_additional = self.get_properties(properties_additional, max_properties=1)
+
+        additional_props = dict(zip(properties_additional, values_additional))
+        data.update(additional_props)
+
+        return WalkingpadStatus(data)
+
+    @command(
+        default_output=format_output(
+            "",
+            "Mode: {result.mode.name}\n"
+            "Time: {result.walking_time}\n"
+            "Steps: {result.step_count}\n"
+            "Speed: {result.speed}\n"
+            "Distance: {result.distance}\n"
+            "Calories: {result.calories}",
+        )
+    )
+    def quick_status(self) -> WalkingpadStatus:
+        """Retrieve quick properties. The walkingpad provides the option.
+
+        to retrieve a subset of properties in one call - steps, mode,
+        speed, distance, calories and time. If you can, use this instead
+        of the full status request, as it is way faster.
+        """
```

Comment:
```suggestion
        """Retrieve quick status. 
        
        The walkingpad provides the option to retrieve a subset of properties in one call:
        steps, mode, speed, distance, calories and time.
        
        `status()` will do four more separate I/O requests for power, mode, start_speed, and sensitivity.
        If you don't need any of that, prefer this method for status updates.
        """
```
Just a suggestion, but the 1st line of docstring should be in any case just "Retrieve quick status" or so without that part from the more elaborate description..

Reply:
Done, good call :)